aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -66,236 +57,344 @@ impl AggregatorID {
     }
 }
 
-/// Tracks values seen by aggregator. In particular, stores information about
-/// the biggest and the smallest deltas seen during execution in the VM. This
-/// information can be used by the executor to check if delta should have
-/// failed. Most importantly, it allows commutativity of adds/subs. Example:
-///
-///
-/// This graph shows how delta of aggregator changed during a single transaction
-/// execution:
-///
-/// ```text
-/// +A ===========================================>
-///            ||
-///          ||||                               +X
-///         |||||  ||||||                    ||||
-///      |||||||||||||||||||||||||          |||||
-/// +0 ===========================================> time
-///                       ||||||
-///                         ||
-///                         ||
-/// -B ===========================================>
-/// ```
-///
-/// Clearly, +X succeeds if +A and -B succeed. Therefore each delta
-/// validation consists of:
-///   1. check +A did not overflow
-///   2. check -A did not drop below zero
-/// Checking +X is irrelevant since +A >= +X.
-///
-/// TODO: while we support tracking of the history, it is not yet fully used on
-/// executor side because we don't know how to throw errors.
-#[derive(Debug)]
-pub struct History {
-    pub max_positive: u128,
-    pub min_negative: u128,
+/// Describes how the `speculative_start_value` in `AggregatorState` was obtained.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SpeculativeStartValue {
+    // The speculative_start_value is assigned to 0.
+    Unset,
+    // The speculative_start_value was obtained by reading
+    // the last committed value of the aggregator from MVHashmap.
```

Comment:
maybe let's add a BIG warning here or in the struct comments, that the LastCommittedValue should only be used for branch prediction purposes (try_add / try_sub), or else with parallel execution we may get incorrect / non-deterministic results.

Reply:
It's not the branch prediction that's important, it is that all consequences leaking to the caller need be in the change set as appropriate restrictions. will add some comments